lib/linked-list.js

```diff
@@ -17,87 +17,180 @@ class LinkedList {
+
+        while(current != null) {
+            if(current.value === value) {
+                return true;
+            }else {
+                current = current.next;
+        }
     }
+    return false; 
+}
 
     /*
     method to return the max value in the linked list
     returns the data value and not the node
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
     findMax() {
-        throw new Error("This method hasn't been implemented yet!");
+        let maxNum = 0 
+
+        let current = this.#head
+        if (current == null) return null
+
+        while(current != null) {
+            if(current.value > maxNum) {
+                maxNum = current.value
+            }else {
+                current = current.next
+            }
+        }
+        return maxNum
     }
     /*
     method to return the min value in the linked list
     returns the data value and not the node
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
      */
     findMin() {
-        throw new Error("This method hasn't been implemented yet!");
+        let current = this.#head
+        if (current == null) return null;
+
+        let minNum = current.value;
+        while(current != null) {
+            if(current.value < minNum) {
+                minNum = current.value
+            }else {
+                current = current.next
+            }
+        }
+        return minNum
     }
 
     /*
     method that returns the length of the singly linked list
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
      */
     length() {
-        throw new Error("This method hasn't been implemented yet!");
+        let current = this.#head
+        if (current == null) return 0;
+        let length = 0
+
+        while(current != null) {
+            length ++;
+            current = current.next;
+        }
+        return length  
     }
 
     /*
     method that returns the value at a given index in the linked list
     index count starts at 0
     returns nil if there are fewer nodes in the linked list than the index value
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
      */
     getAtIndex(index) {
-        throw new Error("This method hasn't been implemented yet!");
+        let current = this.#head
+        let count = 0
+        if (current == null) return null
+        while(count != null && count < index) {
+            count++    
+            current = current.next
+        }   
+        return current.value
     }
 
     /*
     method to print all the values in the linked list
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n) * O(k) -- i don't know! You have to first traverse the linked list and THEN an array
+    Space Complexity: ???
     */
     visit() {
```

Comment:
👍 , since you are traversing the list twice (once for the ll and the other time for the array) it's O(n + n) => O(n) in time complexity.

Since you are creating an array the space complexity is O(n).